### PR TITLE
copy: add support for non-octal mode setting

### PR DESCRIPTION
**Pull request by tonistiigi**
prerequisite for https://github.com/moby/buildkit/issues/1951

### CODE REVIEW

File: bench/go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/tonistiigi/fsutil/bench
 
-go 1.20
+go 1.21
+
+toolchain go1.23.1
```
Comment (crazy-max) [inline bench/go.mod:5]: toolchain strikes again :sweat_smile:

File: copy/copy_windows.go [hunk 2]
```diff
@@ -37,6 +37,10 @@ func getFileSecurityInfo(name string) (*windows.SID, *windows.ACL, error) {
 }
 
 func (c *copier) copyFileInfo(fi os.FileInfo, src, name string) error {
+	if c.modeSet != nil {
+		return errors.Errorf("non-octal mode not supported on windows")
+	}
```
Comment (crazy-max) [inline copy/copy_windows.go:42]: Don't think a module exists for this but I think on Windows we could use `golang.org/x/sys/windows` to set corresponding Windows ACL permissions if we want that in the future (cc @gabriel-samfira)

**Review by crazy-max — APPROVED** [review]

File: bench/go.mod (hunk 1 above)
Comment (tonistiigi) [reply bench/go.mod:5]: :rage1:

**Review by crazy-max — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/opencontainers/go-digest v1.0.0
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.8.4
+	github.com/tonistiigi/dchapes-mode v0.0.0-20241001053921-ca0759fec205
```
Comment (thaJeztah) [inline go.mod:12]: I see the original repository has a number of tagged versions already; should those be inherited? https://hg.sr.ht/~dchapes/mode/tags

Comment (tonistiigi) [reply go.mod:12]: This is not the same source. I don't plan to make my own releases in the fork.

**Pull request merged**